optimize/backend/src/main/java/io/camunda/optimize/rest/PublicApiRestService.java

```diff
@@ -229,6 +232,26 @@ public void disableShare() {
     settingsService.setSettings(settings);
   }
 
+  // Undocumented API used in consulting to programmatically list collections
+  @GET
+  @Path(COLLECTION_SUB_PATH)
+  @Produces(MediaType.APPLICATION_JSON)
+  @SneakyThrows
+  public List<IdResponseDto> getCollectionIds() {
+    return collectionService.getAllCollections().stream()
```

Comment:
Not a blocker but just for your own expectations, this uses a scroll to fetch a response and in environments with a large number of collections, can be a relatively expensive ES/OS query. For your use case, I don't think this is currently an issue and is rather just something to bear in mind